pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -0,0 +1,100 @@
+
+package replication_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/security/securityassets"
+	"github.com/cockroachdb/cockroach/pkg/security/securitytest"
+	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReaderCatalog(t *testing.T) {
+	ctx := context.Background()
+	ts := serverutils.StartServerOnly(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestControlsTenantsExplicitly,
+	})
+	defer ts.Stop(ctx)
+	srcTenant, _, err := ts.StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
+		TenantID:   serverutils.TestTenantID(),
+		TenantName: "src",
+	})
+	require.NoError(t, err)
+	destTenant, _, err := ts.StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
+		TenantID:   serverutils.TestTenantID2(),
+		TenantName: "dest",
+	})
+	require.NoError(t, err)
+	srcConn := srcTenant.SQLConn(t)
+	destConn := destTenant.SQLConn(t)
+
+	ddlToExec := []string{
+		"CREATE USER roacher WITH CREATEROLE;",
+		"GRANT ADMIN TO roacher;",
+		"ALTER USER roacher SET timezone='America/New_York';",
+		"CREATE DATABASE db1;",
+		"CREATE SCHEMA db1.sc1;",
+		"CREATE SEQUENCE sq1;",
+		"CREATE TYPE IF NOT EXISTS status AS ENUM ('open', 'closed', 'inactive');",
+		"CREATE TABLE t1(n int default nextval('sq1'), val status);",
+		"INSERT INTO t1(val) VALUES('open');",
+		"INSERT INTO t1(val) VALUES('closed');",
+		"INSERT INTO t1(val) VALUES('inactive');",
+		"CREATE VIEW v1 AS (SELECT n from t1);",
+	}
+	for _, ddl := range ddlToExec {
+		_, err = srcConn.Exec(ddl)
+		require.NoError(t, err)
+	}
+
+	now := ts.Clock().Now()
+	idb := destTenant.InternalDB().(*sql.InternalDB)
+	require.NoError(t, replication.SetupOrAdvanceStandbyReaderCatalog(ctx, serverutils.TestTenantID(), now, idb, destTenant.ClusterSettings()))
+
+	srcRunner := sqlutils.MakeSQLRunner(srcConn)
+	destRunner := sqlutils.MakeSQLRunner(destConn)
+
+	compareConn := func(query string) {
+		srcRes := srcRunner.QueryStr(t, fmt.Sprintf("SELECT * FROM (%s) AS OF SYSTEM TIME %s", query, now.AsOfSystemTime()))
+		destRes := destRunner.QueryStr(t, query)
+		require.Equal(t, srcRes, destRes)
+	}
+
+	// Validate tables and views match in the catalog reader
+	compareConn("SELECT * FROM t1 ORDER BY n")
+	compareConn("SELECT * FROM v1 ORDER BY 1")
+
+	// Validate that systme tables are synced
+	compareConn("SELECT * FROM system.users")
+	compareConn("SELECT * FROM system.table_statistics")
+	compareConn("SELECT * FROM system.role_options")
+	compareConn("SELECT * FROM system.database_role_settings")
```

Comment:
could you add a few test cases, where you rerun SetupOrAdvanceStandbyReaderCatalog with a new AOST after:
-  insert into t1
- create a new table and insert data into it
- conduct some schema change on it (e.g. add column in it)
- drop that table new table